pkg/frontend/v2/frontend.go

```diff
@@ -205,6 +218,18 @@ func (f *Frontend) RoundTripGRPC(ctx context.Context, req *httpgrpc.HTTPRequest)
 			retryOnTooManyOutstandingRequests: f.cfg.RetryOnTooManyOutstandingRequests && f.schedulerWorkers.getWorkersCount() > 1,
 		}
 
+		queryPriority := f.limits.QueryPriority(userID)
+
+		if reqParams != nil && queryPriority.Enabled {
+			queryPriorityChanged := !reflect.DeepEqual(f.queryPriority, queryPriority)
```

Comment:
`queryPriority` seems per tenant.
But `f.queryPriority` seems a single value used in `Frontend`. Maybe you want to change it to per tenant as well? Otherwise different users will trigger priority change

Reply:
Oh great catch. Addressed in https://github.com/cortexproject/cortex/pull/5605/commits/96ff019ddba591815be280b4f9c9aacf63733a42